src/system/signal/set.rs

```diff
@@ -30,14 +30,16 @@ impl SignalAction {
             }
         };
 
-        Ok(Self {
-            raw: libc::sigaction {
-                sa_sigaction,
-                sa_mask: sa_mask.raw,
-                sa_flags,
-                sa_restorer: None,
-            },
-        })
+        // SAFETY: since sigaction is a C struct, all-zeroes is a valid representation
+        // We cannot use a "literal struct" initialization method since the exact representation
+        // of libc::sigaction is not fixed, see e.g. https://github.com/memorysafety/sudo-rs/issues/829
+        let mut raw: libc::sigaction = unsafe { std::mem::zeroed() };
```

Comment:
I'd prefer to put this in a function to reduce our usage of the unsafe keyword.

Reply:
Those don't necessarily worry me too much, but I'm fine with that change (be aware that this trick needs to happen in two rather separate files, so be strategic about where you put it)